ProductWarehouse.UnitTests/InfrastructureTests/ProductRepositoryTests.cs

```diff
@@ -0,0 +1,111 @@
+﻿using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+using Moq;
+using Moq.Protected;
+using Newtonsoft.Json;
+using ProductWarehouse.Domain.Entities;
+using ProductWarehouse.Infrastructure.Http;
+using ProductWarehouse.Infrastructure.Repositories;
+using System.Net;
+using Xunit;
+
+public class ProductRepositoryTests
+{
+    [Fact]
+    public async Task GetProductsAsync_ReturnsEmptyList()
+    {
+        // Arrange
+        string testLink = "https://testlink";
```

Comment:
this is not a valid link  I don't see the top domain level(.bg, .com, .tm, .net, .org end etc.). Changed to  "https://testlink.com";